instrumentation/spring/spring-security-config-6.0/library/src/main/java/io/opentelemetry/instrumentation/spring/security/config/v6_0/EnduserAttributesCapturer.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.security.config.v6_0;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.LocalRootSpan;
+import io.opentelemetry.semconv.SemanticAttributes;
+import java.util.Objects;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+
+/**
+ * Captures {@code enduser.*} semantic attributes from {@link Authentication} objects.
+ *
+ * <p>After construction, you must selectively enable which attributes you want captured by calling
+ * the appropriate {@code set*Enabled(true)} method.
+ */
+public class EnduserAttributesCapturer {
```

Comment:
Our convention is to make classes that are intended to be used outside of this project `final`. If a library class is not intended to be used directly we make it package private or move it to `internal` package.

Reply:
Made the class final.

I intended to allow this class to be used directly, because the agent only instruments inbound HTTP requests, whereas applications could have other types of inbound requests for which they want to capture enduser attributes.